glocal_exploration_ros/src/glocal_system.cpp

```diff
@@ -112,10 +112,20 @@ void GlocalSystem::mainLoop() {
   run_srv_ = nh_private_.advertiseService("toggle_running",
                                           &GlocalSystem::runSrvCallback, this);
 
+  // Limit the maximum update frequency.
+  // NOTE: This is mainly intended to avoid spinning at unlimited rates in case
+  //       a loopIteration() returns immediately. For example, when the global
+  //       planner is idling while waiting for the next waypoint to be reached.
+  ros::Rate max_rate(100 /*Hz*/);
+
+  // Spin.
   while (ros::ok() && comm_->stateMachine()->currentState() !=
                           StateMachine::State::kFinished) {
     loopIteration();
     ros::spinOnce();
+
+    // Sleep only if the maximum rate would otherwise be exceeded.
+    max_rate.sleep();
```

Comment:
I didn't know ros::Rate can sleep excess time for you, good find :)

Reply:
Do the local planner iterations sometimes take less than 0.01s? Otherwise, rate.sleep() would always immediately return.

Reply:
Yes, e.g. as a sampled point is infeasible the local planner will start over, i.e. it can return within few lines to re-start sampling, which runs (until gain computation) at few 1000 hz if I recall correctly. Using a max-rate of 100 will potentially not be a large impact in most cases but if the value was to be set lower or in specific circumstances this might be relevant.

Reply:
Ok, I exempted the local planner from the rate limit and added some explanatory comments.